ksqldb-engine/src/main/java/io/confluent/ksql/query/MaterializationProviderBuilderFactory.java

```diff
@@ -77,22 +77,23 @@ public MaterializationProviderBuilder materializationProviderBuilder(
         keyFormat,
         streamsProperties,
         applicationId,
-        queryId
+        queryId,
+        pullQueryKeyFormat
     );
   }
 
   private Optional<MaterializationProvider> buildMaterializationProvider(
-      final KafkaStreams kafkaStreams,
-      final Topology topology,
-      final MaterializationInfo materializationInfo,
-      final PhysicalSchema schema,
-      final KeyFormat keyFormat,
-      final Map<String, Object> streamsProperties,
-      final String applicationId,
-      final String queryId
-  ) {
+          final KafkaStreams kafkaStreams,
+          final Topology topology,
+          final MaterializationInfo materializationInfo,
+          final PhysicalSchema schema,
+          final KeyFormat keyFormat,
```

Comment:
I don't think you need a separate `keyFormat` for the `keyFormat.getWindowInfo()`, do you -- can you just use the same value as `pullQueryKeyFormat`?  When I was saying you could remove `pullQueryKeyFormat` earlier, I thought you could remove it all the way to the top where you're deciding whether or not a special case is needed.  I'll defer to you on this, since you've tested it, but it's confusing when `keyFormat` is used vs `pullQueryKeyFormat`.  They're all for pull queries! (That's the only functionality that uses this materialization provider)